test-suite/squidconf/tls-certs

```diff
@@ -0,0 +1,256 @@
+## Copyright (C) 1996-2018 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+#
+# The following tests check the sparse matrix of (valid) configs:
+#
+#  A) tls-cert PEM contents:
+#    1) cert only           tls-cert=./leaf-key-rsa.pem
+#    2) full chain          tls-cert=./leaf-chain-rsa.pem
+#    3) root CA             tls-cert=./ca-root-rsa.pem
+#    4) incomplete chain    tls-cert=./ca-mid-rsa.pem
+#    5) missing
+#
+#  B) tls-key parameter:
+#    1) separate:       tls-cert=./leaf-rsa.crt tls-key=./leaf-rsa.pkey
+#    2) merged:         tls-cert=./leaf-key-rsa.pem
+#
+#  C) tls-generate-host-certificates values:
+#    1) missing
+#    2) on:         tls-generate-host-certificates=on
+#    3) off:        tls-generate-host-certificates=off
+#    4) auto        tls-generate-host-certificates
```

Comment:
done

Reply:
> done

Please push the changes _before_ commenting on them or marking the corresponding change requests as resolved. Git commits and pushes are cheap. The opposite "annotate everything based on what I see in my private repository copy and then push" order is more likely to confuse the reviewer and/or waste their time because the reviewer is likely to react to annotations that do not match the code.

Reply:
Hmm. Somehow I lost the commit doing the documentation updates. Re-done.